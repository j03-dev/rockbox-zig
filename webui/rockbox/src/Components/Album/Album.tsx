/* eslint-disable @typescript-eslint/no-explicit-any */
import { FC } from "react";
import {
  AlbumCover,
  AlbumFooterMenu,
  AlbumTitle,
  Artist,
  FloatingButton,
  Hover,
  Link,
  NoAlbumCover,
  Year,
} from "./styles";
import Play from "../Icons/Play";
import ContextMenu from "./ContextMenu";
import HeartOutline from "../Icons/HeartOutline";
import AlbumArt from "../../Assets/albumart.svg";

export type AlbumProps = {
  album: any;
  onPlay: (album: any) => void;
  onLike: (album: any) => void;
  onUnLike: (album: any) => void;
};

const Album: FC<AlbumProps> = (props) => {
  return (
    <div style={{ position: "relative", width: "100%" }}>
      <Hover>
        <AlbumFooterMenu>
          <div
            style={{
              backgroundColor: "#ffffffda",
              height: 40,
              width: 40,
              borderRadius: 20,
              display: "flex",
              justifyContent: "center",
              alignItems: "center",
            }}
            onClick={() => props.onPlay(props.album)}
          >
            <Play small color="#000" />
          </div>
          <ContextMenu item={props.album} />
          <FloatingButton onClick={() => props.onLike(props.album)}>
            <HeartOutline color="#fff" size={20} />
          </FloatingButton>
        </AlbumFooterMenu>
      </Hover>
      <Link to={`/albums/${props.album.id}`}>
        {props.album.cover && (
          <AlbumCover src={props.album.cover} effect="opacity" />
        )}
        {!props.album.cover && <NoAlbumCover src={AlbumArt} />}
        <AlbumTitle>{props.album.title}</AlbumTitle>
      </Link>
      <Artist to={`/artists/${props.album.artistId}`}>
        {props.album.artist}
      </Artist>
      <Year>{props.album.year}</Year>
    </div>
  );
};

export default Album;
