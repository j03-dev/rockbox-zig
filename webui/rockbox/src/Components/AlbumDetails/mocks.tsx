export const tracks = [
  {
    id: "1",
    trackNumber: 1,
    title: "Gasoline (Live)",
    artist: "Seether",
    time: "2:57",
  },
  {
    id: "2",
    trackNumber: 2,
    title: "Driven Under (Live)",
    artist: "Seether",
    time: "4:58",
  },
  {
    id: "3",
    trackNumber: 3,
    title: "Diseased (Live)",
    artist: "Seether",
    time: "3:46",
  },
  {
    id: "4",
    trackNumber: 4,
    title: "Truth (Live)",
    artist: "Seether",
    time: "5:15",
  },
  {
    id: "5",
    trackNumber: 5,
    title: "Immortality (Live)",
    artist: "Seether",
    time: "5:02",
  },
  {
    id: "6",
    trackNumber: 6,
    title: "Tied My Hands (Live)",
    artist: "Seether",
    time: "5:16",
  },
  {
    id: "7",
    trackNumber: 7,
    title: "Sympathetic (Live)",
    artist: "Seether",
    time: "4:12",
  },
  {
    id: "8",
    trackNumber: 8,
    title: "Fine Again (Live)",
    artist: "Seether",
    time: "5:05",
  },
  {
    id: "9",
    trackNumber: 9,
    title: "Broken (Live)",
    artist: "Seether",
    time: "4:17",
  },
  {
    id: "10",
    trackNumber: 10,
    title: "The Gift (Live)",
    artist: "Seether",
    time: "5:36",
  },
  {
    id: "11",
    trackNumber: 11,
    title: "Remedy (Live)",
    artist: "Seether",
    time: "3:40",
  },
  {
    id: "12",
    trackNumber: 12,
    title: "Plastci Man (Live)",
    artist: "Seether",
    time: "3:34",
  },
  {
    id: "13",
    trackNumber: 13,
    title: "The Gist (Alternate Mix)",
    artist: "Seether",
    time: "3:24",
  },
];
