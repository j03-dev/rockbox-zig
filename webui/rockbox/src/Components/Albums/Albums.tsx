/* eslint-disable @typescript-eslint/no-explicit-any */
import { FC } from "react";
import { Cell, Grid } from "baseui/layout-grid";
import MainView from "../MainView";
import Sidebar from "../Sidebar";
import ControlBar from "../ControlBar";
import AlbumArt from "../../Assets/albumart.svg";
import {
  AlbumCover,
  AlbumFooterMenu,
  AlbumTitle,
  Artist,
  Container,
  FilterContainer,
  FloatingButton,
  Hover,
  Scrollable,
  Title,
  Year,
  Link,
  NoAlbumCover,
} from "./styles";
import Filter from "../Filter";
import Play from "../Icons/Play";
import HeartOutline from "../Icons/HeartOutline";
import ContextMenu from "./ContextMenu";

export type AlbumsProps = {
  albums: any[];
  onFilter: (filter: string) => void;
  onPlay: (album: any) => void;
  onLike: (album: any) => void;
  onUnLike: (album: any) => void;
};

const Albums: FC<AlbumsProps> = (props) => {
  const { albums } = props;
  return (
    <Container>
      <Sidebar active="albums" />
      <MainView>
        <ControlBar />
        <Scrollable>
          <Title>Albums</Title>
          <FilterContainer>
            <Filter placeholder="Search albums" onChange={() => {}} />
          </FilterContainer>
          <div style={{ marginBottom: 100 }}>
            <Grid
              gridColumns={[2, 4, 5]}
              gridMargins={[20, 20, 20]}
              gridGutters={[25, 25, 25]}
            >
              {albums.map((item) => (
                <Cell key={item.id}>
                  <div style={{ position: "relative", width: "100%" }}>
                    <Hover>
                      <AlbumFooterMenu>
                        <div
                          style={{
                            backgroundColor: "#ffffffda",
                            height: 40,
                            width: 40,
                            borderRadius: 20,
                            display: "flex",
                            justifyContent: "center",
                            alignItems: "center",
                          }}
                          onClick={() => props.onPlay(item)}
                        >
                          <Play small color="#000" />
                        </div>
                        <ContextMenu
                          album={item}
                          onPlayNext={() => {}}
                          onCreatePlaylist={() => {}}
                          onAddTrackToPlaylist={() => {}}
                          onPlayLast={() => {}}
                          recentPlaylists={[]}
                        />
                        <FloatingButton onClick={() => props.onLike(item)}>
                          <HeartOutline color="#fff" size={20} />
                        </FloatingButton>
                      </AlbumFooterMenu>
                    </Hover>
                    <Link to={`/albums/${item.id}`}>
                      {item.cover && (
                        <AlbumCover src={item.cover} effect="opacity" />
                      )}
                      {!item.cover && <NoAlbumCover src={AlbumArt} />}
                      <AlbumTitle>{item.title}</AlbumTitle>
                    </Link>
                    <Artist to={`/artists/${item.artistId}`}>
                      {item.artist}
                    </Artist>
                    <Year>{item.year}</Year>
                  </div>
                </Cell>
              ))}
            </Grid>
          </div>
        </Scrollable>
      </MainView>
    </Container>
  );
};

export default Albums;
