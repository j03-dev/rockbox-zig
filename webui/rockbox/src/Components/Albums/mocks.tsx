export const albums = [
  {
    id: "1",
    title: "The End, So Far",
    artist: "Slipknot",
    cover:
      "https://resources.tidal.com/images/b80f3852/73e5/43d6/90ea/021d1fad2bbe/320x320.jpg",
    year: 2022,
  },
  {
    id: "2",
    title: "One Cold Night (Live)",
    artist: "Seether",
    cover:
      "https://resources.tidal.com/images/f6f5f0a6/dc95/4561/9ca6/6ba1e0f6a062/320x320.jpg",
    year: 2006,
  },
  {
    id: "3",
    title: "4Eva N A Day",
    artist: "Big K.R.I.T.",
    cover:
      "https://resources.tidal.com/images/3d58ba7f/8e6d/4a2e/a2e4/793495b8916a/320x320.jpg",
    year: 2012,
  },
  {
    id: "4",
    title: "Pluto x Baby Pluto (Deluxe)",
    artist: "Future, Lil Uzi Vert",
    cover:
      "https://resources.tidal.com/images/fe6787d5/4ba5/4d3e/8576/48943ee6a768/320x320.jpg",
    year: 2020,
  },
  {
    id: "5",
    title: "The Acoustic Sessions, Vol. 2",
    artist: "Smile Empty Soul",
    cover:
      "https://resources.tidal.com/images/06a15286/8b80/40a5/bd40/187e99e2e80d/320x320.jpg",
    year: 2020,
  },
  {
    id: "6",
    title: "Black Summer",
    artist: "Red Hot Chilli Peppers",
    cover:
      "https://resources.tidal.com/images/0bb1f23f/94a3/4211/8b6a/be57904ea709/320x320.jpg",
    year: 2022,
  },
  {
    id: "7",
    title: "Californication (Deluxe Edition)",
    artist: "Red Hot Chilli Peppers",
    cover:
      "https://resources.tidal.com/images/543575fc/ad02/419b/ae61/671558dc019d/320x320.jpg",
    year: 1999,
  },
  {
    id: "8",
    title: "Ghetto Lenny's Love Songs",
    artist: "SAINt JHN",
    cover:
      "https://resources.tidal.com/images/f2d921e6/6723/4a7c/a3c0/f630c9ce94eb/320x320.jpg",
    year: 2019,
  },
  {
    id: "9",
    title: "Memphis Zoo",
    artist: "Juicy J, Xavier Wulf",
    cover:
      "https://resources.tidal.com/images/982470d8/9c3e/476d/ae2d/ecac2c555d09/320x320.jpg",
    year: 2024,
  },
  {
    id: "10",
    title: "Mental Trillness 2",
    artist: "Juicy J",
    cover:
      "https://resources.tidal.com/images/e4469204/8f6d/47ca/b98a/04388ec47fdc/320x320.jpg",
    year: 2024,
  },
  {
    id: "11",
    title: "Live From The Underground",
    artist: "Big K.R.I.T.",
    cover:
      "https://resources.tidal.com/images/2f78cf3f/79c8/4ad6/9d12/f03678768a31/320x320.jpg",
    year: 2012,
  },
  {
    id: "12",
    title: "One Foot Out The Door",
    artist: "Mike Posner",
    cover:
      "https://resources.tidal.com/images/4052a2df/fc79/4b9b/9d1c/0507e2c270bf/1280x1280.jpg",
    year: 2009,
  },
  {
    id: "13",
    title: "Rockstar Mentality",
    artist: "Shop Boyz",
    cover:
      "https://resources.tidal.com/images/e9bd1318/b327/4c69/abc4/5bcb33ec5a31/320x320.jpg",
    year: 2007,
  },
  {
    id: "13",
    title: "The L.I.B.R.A.",
    artist: "T.I.",
    cover:
      "https://resources.tidal.com/images/5d6aa2b6/622d/4df2/82d2/d80d26f49f1d/320x320.jpg",
    year: 2020,
  },
  {
    id: "14",
    title: "Young and Turnt",
    artist: "42 Dugg",
    cover:
      "	https://resources.tidal.com/images/67d5a152/c200/42a2/9332/bf3b301daa38/320x320.jpg",
    year: 2019,
  },
  {
    id: "15",
    title: "4eva Us Neva Them",
    artist: "42 Dugg",
    cover:
      "https://resources.tidal.com/images/159d4317/1d58/45f3/97a0/44595b498f69/320x320.jpg",
    year: 2024,
  },
  {
    id: "16",
    title: "Post Traumatic",
    artist: "Tee Grizzley",
    cover:
      "https://resources.tidal.com/images/6d86e202/63f5/487b/9f38/0113afef3912/320x320.jpg",
    year: 2024,
  },
  {
    id: "17",
    title: "King",
    artist: "T.I.",
    cover:
      "https://resources.tidal.com/images/636be449/9e78/4adc/92a0/84f11cbc658a/320x320.jpg",
    year: 2006,
  },
  {
    id: "18",
    title: "Meteora 20th Anniversary Edition",
    artist: "Linkin Park",
    cover:
      "https://resources.tidal.com/images/ac1e1f54/4f1c/491e/a373/369352f887e8/320x320.jpg",
    year: 2023,
  },
  {
    id: "19",
    title: "The Surface Seems So Far",
    artist: "Seether",
    cover:
      "https://resources.tidal.com/images/0a30cfd9/5b9b/4fd4/b172/a4c625ce1dbb/320x320.jpg",
    year: 2024,
  },
  {
    id: "20",
    title: "Karma and Effect",
    artist: "Seether",
    cover:
      "https://resources.tidal.com/images/8a3da74c/4e05/4ded/afb1/a6bb19855952/320x320.jpg",
    year: 2005,
  },
  {
    id: "21",
    title: "Disclaimer",
    artist: "Seether",
    cover:
      "https://resources.tidal.com/images/33ac8699/ee88/46ac/8156/f508757d0128/320x320.jpg",
    year: 2002,
  },
  {
    id: "21",
    title: "Disclaimer II",
    artist: "Seether",
    cover:
      "https://resources.tidal.com/images/f3f10be8/d97d/4ace/acb5/c7460d9727e6/320x320.jpg",
    year: 2002,
  },
  {
    id: "22",
    title: "Fallen (Deluxe Edition / Remastered 2023)",
    artist: "Evanesence",
    cover:
      "https://resources.tidal.com/images/898cc800/6c0a/4c7b/82f6/6b3be9527e7b/320x320.jpg",
    year: 2003,
  },
  {
    id: "23",
    title: "Finding Beauty In Negative Spaces (Bonus Track Version)",
    artist: "Seether",
    cover:
      "https://resources.tidal.com/images/00ac69d0/79db/446e/991c/72436c74a460/320x320.jpg",
    year: 2007,
  },
  {
    id: "24",
    title: "Meteora (Bonus Edition)",
    artist: "Linkin Park",
    cover:
      "https://resources.tidal.com/images/06bc5258/a07e/480a/b83a/1c5b4857b622/320x320.jpg",
    year: 2003,
  },
  {
    id: "25",
    title: "Hybrid Theory (Deluxe Edition)",
    artist: "Linkin Park",
    cover:
      "https://resources.tidal.com/images/2384f908/261a/46a6/a100/165353b2fb63/320x320.jpg",
    year: 2003,
  },
  {
    id: "26",
    title: "T.I. vs T.I.P.",
    artist: "T.I.",
    cover:
      "https://resources.tidal.com/images/fa5b761e/996f/4fef/8a9b/2eff914925f5/320x320.jpg",
    year: 2007,
  },
  {
    id: "27",
    title: "Paper Trail",
    artist: "T.I.",
    cover:
      "https://resources.tidal.com/images/3388ff3c/e69a/4d3b/94f0/e9f36d0adb0c/320x320.jpg",
    year: 2008,
  },
  {
    id: "28",
    title: "No Mercy",
    artist: "T.I.",
    cover:
      "https://resources.tidal.com/images/d0cec70c/a52e/4545/872f/aa6a0d0dd4e1/320x320.jpg",
    year: 2010,
  },
  {
    id: "29",
    title: "Paperwork (Explicit)",
    artist: "T.I.",
    cover:
      "https://resources.tidal.com/images/603de40f/ac1d/4d5f/be74/9ff06f8eb9f5/320x320.jpg",
    year: 2014,
  },
  {
    id: "30",
    title: "Paperwork (Deluxe Explicit)",
    artist: "T.I.",
    cover:
      "https://resources.tidal.com/images/e6e44784/1117/4e80/9233/0a13488e3b0c/320x320.jpg",
    year: 2014,
  },
  {
    id: "31",
    title: "The Bluez Brothaz",
    artist: "Bluez Brothaz, T-Pain, Young Cash",
    cover:
      "https://resources.tidal.com/images/5d2e6bed/c2c1/48e6/8aa5/cf12d7c0ba52/1280x1280.jpg",
    year: 2024,
  },
  {
    id: "32",
    title: "Pimpalation (Limited Edition)",
    artist: "Pimp C",
    cover:
      "https://resources.tidal.com/images/ea2f9c73/6bfc/4dd5/be15/163062d66dea/320x320.jpg",
    year: 2006,
  },
  {
    id: "33",
    title: "Long Live The Pimp",
    artist: "Pimp C",
    cover:
      "https://resources.tidal.com/images/3818842b/a19f/4aa7/8568/23f7bdab718a/320x320.jpg",
    year: 2015,
  },
  {
    id: "33",
    title: "The Naked Soul of Sweet Jones",
    artist: "Pimp C",
    cover:
      "https://resources.tidal.com/images/c85170de/f63c/4f5f/a006/3a1e34369f55/1280x1280.jpg",
    year: 2010,
  },
  {
    id: "34",
    title: "Epiphany (Expanded Edition)",
    artist: "T-Pain",
    cover:
      "https://resources.tidal.com/images/64688543/db20/444b/a0a8/607a7b36f3fa/320x320.jpg",
    year: 2007,
  },
  {
    id: "35",
    title: "Mental Trillness",
    artist: "Juicy J",
    cover:
      "https://resources.tidal.com/images/625de685/5ec5/434b/bdd7/10d66e9aadec/320x320.jpg",
    year: 2023,
  },
  {
    id: "36",
    title: "Class In Session: The Boyfriend/Girlfriend EP",
    artist: "C-Side",
    cover:
      "https://resources.tidal.com/images/056940dc/3f1a/4c39/8bda/290c82f9e1e4/320x320.jpg",
    year: 2008,
  },
  {
    id: "37",
    title: "TURQUOiSE TORNADO",
    artist: "Riff Raff, Yelawolf",
    cover:
      "https://resources.tidal.com/images/e2a0c659/2348/4dd6/97e3/14d64ea454c1/1280x1280.jpg",
    year: 2021,
  },
];
