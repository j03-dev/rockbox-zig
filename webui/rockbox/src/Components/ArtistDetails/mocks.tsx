export const tracks = [
  {
    id: "1",
    trackNumber: 1,
    title: "Get Lucky",
    artist: "Daft Punk",
    album: "Random Access Memories",
    albumArt:
      "https://resources.tidal.com/images/09b59e6e/717e/43e3/b2e2/d2a153c24775/320x320.jpg",
    time: "6:10",
    artistId: "1",
    albumId: "1",
  },
  {
    id: "2",
    trackNumber: 2,
    title: "Instant Crush",
    artist: "Daft Punk",
    album: "Random Access Memories",
    albumArt:
      "https://resources.tidal.com/images/09b59e6e/717e/43e3/b2e2/d2a153c24775/320x320.jpg",
    time: "5:38",
    artistId: "1",
    albumId: "2",
  },
  {
    id: "3",
    trackNumber: 2,
    title: "Arround the World",
    artist: "Daft Punk",
    album: "Homework",
    albumArt:
      "https://resources.tidal.com/images/f853861c/0c5f/4e73/b608/eeb00618fe6f/320x320.jpg",
    time: "7:10",
    artistId: "1",
    albumId: "3",
  },
];

export const albums = [
  {
    id: "1",
    title: "Random Access Memories",
    artist: "Daft Punk",
    albumArt:
      "https://resources.tidal.com/images/09b59e6e/717e/43e3/b2e2/d2a153c24775/320x320.jpg",
    year: 2013,
  },
  {
    id: "2",
    title: "Tron: Legacy",
    artist: "Daft Punk",
    albumArt:
      "https://resources.tidal.com/images/866bb671/5ec8/4b45/8a60/50e51f5dbf10/320x320.jpg",
    year: 2010,
  },
  {
    id: "3",
    title: "Homework",
    artist: "Daft Punk",
    albumArt:
      "https://resources.tidal.com/images/f853861c/0c5f/4e73/b608/eeb00618fe6f/320x320.jpg",
    year: 2001,
  },
];
