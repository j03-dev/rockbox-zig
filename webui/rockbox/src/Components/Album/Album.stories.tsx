import type { Meta, StoryObj } from "@storybook/react";

import Album from "./Album";
import { albums } from "../Albums/mocks";
import { fn } from "@storybook/test";

// More on how to set up stories at: https://storybook.js.org/docs/writing-stories#default-export
const meta = {
  title: "Components/Album",
  component: Album,
  // This component will have an automatically generated Autodocs entry: https://storybook.js.org/docs/writing-docs/autodocs
  tags: ["autodocs"],
  // More on argTypes: https://storybook.js.org/docs/api/argtypes
} satisfies Meta<typeof Album>;

export default meta;
type Story = StoryObj<typeof meta>;

// More on writing stories with args: https://storybook.js.org/docs/writing-stories/args
export const Default: Story = {
  args: {
    onPlay: fn(),
    onLike: fn(),
    onUnLike: fn(),
    album: albums[0],
  },
};
