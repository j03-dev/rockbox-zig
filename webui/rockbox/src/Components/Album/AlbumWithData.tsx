/* eslint-disable @typescript-eslint/no-explicit-any */
import { FC } from "react";
import Album from "./Album";
import { usePlayAlbumMutation } from "../../Hooks/GraphQL";

export type AlbumWithDataProps = {
  album: any;
};

const AlbumWithData: FC<AlbumWithDataProps> = ({ album }) => {
  const [playAlbum] = usePlayAlbumMutation();

  const onPlay = ({ id: albumId }: any) => {
    playAlbum({
      variables: {
        albumId,
      },
    });
  };
  return (
    <Album
      album={album}
      onLike={() => {}}
      onPlay={onPlay}
      onUnLike={() => {}}
    />
  );
};

export default AlbumWithData;
