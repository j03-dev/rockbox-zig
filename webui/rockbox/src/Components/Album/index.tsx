import Album from "./AlbumWithData";

export default Album;
