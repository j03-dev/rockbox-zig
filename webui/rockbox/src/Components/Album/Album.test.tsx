import { render } from "@testing-library/react";
import { vi } from "vitest";
import Album from "./Album";
import { albums } from "../Albums/mocks";
import Providers from "../../Providers";
import { MemoryRouter } from "react-router-dom";
import { MockedProvider } from "@apollo/client/testing";
import { mocks } from "../../mocks";

describe("Album", () => {
  it("should render", () => {
    const onPlay = vi.fn();
    const onUnLike = vi.fn();
    const onLike = vi.fn();
    const { container } = render(
      <MemoryRouter initialEntries={["/"]}>
        <MockedProvider mocks={mocks}>
          <Providers>
            <Album
              album={albums[0]}
              onPlay={onPlay}
              onUnLike={onUnLike}
              onLike={onLike}
            />
          </Providers>
        </MockedProvider>
      </MemoryRouter>
    );
    expect(container).toMatchSnapshot();
  });
});
