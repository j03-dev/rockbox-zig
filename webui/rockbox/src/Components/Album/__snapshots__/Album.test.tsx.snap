// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`Album > should render 1`] = `
<div>
  <div
    class=""
  >
    <div
      style="position: relative; width: 100%;"
    >
      <button
        class="css-12ezcoy"
      >
        <div
          class="css-15yrfky"
        >
          <div
            style="background-color: rgba(255, 255, 255, 0.855); height: 40px; width: 40px; border-radius: 20px; display: flex; justify-content: center; align-items: center;"
          >
            <svg
              fill="none"
              height="20"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <path
                class="ionicon"
                d="M7.386 16c-.23 0-.456-.057-.656-.165-.45-.24-.73-.706-.73-1.213V4.378a1.387 1.387 0 0 1 2.071-1.197l9.296 5.241c.394.232.633.64.633 1.077 0 .438-.239.845-.633 1.078L8.07 15.819a1.39 1.39 0 0 1-.684.181Z"
                style="fill: #000;"
              />
            </svg>
          </div>
          <div
            class="css-113iyi7"
          >
            <button
              class="css-10bvv2v"
            >
              <div
                aria-expanded="false"
                aria-haspopup="true"
                class="css-10n9a0h"
              >
                <svg
                  aria-hidden="true"
                  class="StyledIconBase-sc-ea9ulj-0 dmvaRK"
                  color="#fff"
                  fill="currentColor"
                  focusable="false"
                  height="24"
                  viewBox="0 0 512 512"
                  width="24"
                  xmlns="http://www.w3.org/2000/svg"
                >
                  <circle
                    cx="256"
                    cy="256"
                    r="48"
                  />
                  <circle
                    cx="416"
                    cy="256"
                    r="48"
                  />
                  <circle
                    cx="96"
                    cy="256"
                    r="48"
                  />
                </svg>
              </div>
            </button>
          </div>
          <button
            class="css-10n9a0h"
          >
            <svg
              fill="none"
              height="20"
              style="padding-top: 1px;"
              width="20"
              xmlns="http://www.w3.org/2000/svg"
            >
              <g
                class="ionicon"
                style="fill: #fff;"
              >
                <path
                  d="M13.534 4C11.167 4 10 6.364 10 6.364S8.833 4 6.466 4C4.543 4 3.02 5.63 3 7.575c-.04 4.038 3.162 6.91 6.672 9.323a.578.578 0 0 0 .656 0c3.51-2.413 6.712-5.285 6.672-9.323C16.98 5.63 15.457 4 13.534 4Z"
                  fill="none"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  style="fill: none;"
                />
                <path
                  class="stroke-shape"
                  d="M13.534 4C11.167 4 10 6.364 10 6.364S8.833 4 6.466 4C4.543 4 3.02 5.63 3 7.575c-.04 4.038 3.162 6.91 6.672 9.323a.578.578 0 0 0 .656 0c3.51-2.413 6.712-5.285 6.672-9.323C16.98 5.63 15.457 4 13.534 4Z"
                  stroke="currentColor"
                  stroke-linecap="round"
                  stroke-linejoin="round"
                  style="fill: none; stroke-width: 2; stroke: #fff; stroke-opacity: 1;"
                />
              </g>
            </svg>
          </button>
        </div>
      </button>
      <a
        class="css-11g9kr1"
        href="/albums/1"
      >
        <span
          class=" lazy-load-image-background opacity"
          style="color: transparent; display: inline-block;"
        >
          <img
            class="css-1g7adp4"
            src="https://resources.tidal.com/images/b80f3852/73e5/43d6/90ea/021d1fad2bbe/320x320.jpg"
          />
        </span>
        <div
          class="css-e68bk5"
        >
          The End, So Far
        </div>
      </a>
      <a
        class="css-9sbfuc"
        href="/artists/1"
      >
        Slipknot
      </a>
      <div
        class="css-1q570sn"
      >
        2022
      </div>
    </div>
  </div>
  <div
    class=""
  />
</div>
`;
